jupyterhub/auth.py

```diff
@@ -144,6 +192,83 @@ def _deprecated_db(self):
         """
     ).tag(config=True)
 
+    allow_all = Bool(
+        False,
+        config=True,
+        help="""
+        Allow every user who can successfully authenticate to access JupyterHub.
+        
+        False by default, which means for most Authenticators,
+        _some_ allow-related configuration is required to allow any users to log in.
+
+        Authenticator subclasses may override the default with e.g.::
```

Comment:
Are you mentioning this just for backwards compatibility? If so I think it's worth saying that, for new authenticators is it better for them to require explicit configuration?

Reply:
I don't know that it's better! I think it's for Authenticators to decide what makes sense, so I'd leave this as a statement of fact that it _may_ be overridden.